### PR TITLE
doc: Add Doxygen documentation to cdb

### PR DESCRIPTION
**Pull request by createyourpersonalaccount**
The source and header file of cdb is documented. There is also documentation added for the private sdb header which contains helpful macros and static functions.

 

**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [X] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/util/sdb/src/sdb_private.h [hunk 1]
```diff
@@ -35,6 +52,11 @@ static inline void ut32_pack(char s[4], ut32 u) {
 	s[3] = u >> 8;
 }
 
+/**
```
Comment (XVilka) [inline librz/util/sdb/src/sdb_private.h]: In fact, these functiona should probably removed in favor of functions from `librz/include/rz_endian.h`
Now, that SDB is inlined, we can reuse functions from other parts of RzUtil whenever possible.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/util/sdb/src/sdb_private.h (hunk 1 above)
Comment (createyourpersonalaccount) [reply librz/util/sdb/src/sdb_private.h]: Do I just include `<rz_endian.h>`? See new commit.

File: librz/util/sdb/src/cdb_make.c [hunk 2]
```diff
@@ -168,8 +169,8 @@ int cdb_make_finish(struct cdb_make *c) {
 			c->hash[where] = *hp++;
 		}
 		for (u = 0; u < len; u++) {
-			ut32_pack(buf, c->hash[u].h);
-			ut32_pack(buf + 4, c->hash[u].p);
+			rz_write_le32(buf, c->hash[u].h);
+			rz_write_at_le43(buf, c->hash[u].p, 4);
```
Comment (XVilka) [inline librz/util/sdb/src/cdb_make.c]: A typo?

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/util/sdb/src/sdb_private.h (hunk 1 above)
Comment (XVilka) [reply librz/util/sdb/src/sdb_private.h]: Yes.

**Comment by XVilka** [comment]
By the way, while you are on it, you might also swap `cdb_alloc()` and `cdb_alloc_free()` with  `rz_malloc_aligned()` and `rz_free_aligned()` respectively.

**Comment by XVilka** [comment]
And remove `R_ANEW()` macro since it's used only once.

**Comment by createyourpersonalaccount** [comment]
> And remove `R_ANEW()` macro since it's used only once.

I made it be used 3 times. It's okay because it means I don't have to repeat `ALIGNMENT`. What do you think?

File: librz/util/sdb/src/cdb_make.c [hunk 3]
```diff
@@ -3,36 +3,15 @@
...
-}
-
-void cdb_alloc_free(void *x) {
-#if __WINDOWS__ && !__CYGWIN__
-	_aligned_free(x);
-#else
-	free(x);
-#endif
-}
+#define R_ANEW(x, n) (x *)rz_malloc_aligned(n*sizeof(x), ALIGNMENT)
```
Comment (wargio) [inline librz/util/sdb/src/cdb_make.c]:
```suggestion
#define RZ_ANEW(x, n) (x *)rz_malloc_aligned(n*sizeof(x), ALIGNMENT)
```

**Review by wargio — CHANGES_REQUESTED** [review]
.

**Review by XVilka — CHANGES_REQUESTED** [review]
Please run `sys/clang-format.py`:
```
librz/util/sdb/src/cdb_make.c:13:18: error: code should be clang-formatted [-Wclang-format-violations]
#define ALIGNMENT sizeof(void *)
                 ^
librz/util/sdb/src/cdb_make.c:14:47: error: code should be clang-formatted [-Wclang-format-violations]
#define RZ_ANEW(x, n) (x *)rz_malloc_aligned(n*sizeof(x), ALIGNMENT)
                                              ^
librz/util/sdb/src/cdb_make.c:14:48: error: code should be clang-formatted [-Wclang-format-violations]
#define RZ_ANEW(x, n) (x *)rz_malloc_aligned(n*sizeof(x), ALIGNMENT)
                                               ^
```

**Comment by XVilka** [comment]
Please also rebase on top of the latest `dev` since I changed some SDB files

**Comment by createyourpersonalaccount** [comment]
@XVilka I will do this tomorrow.

**Review by wargio — APPROVED** [review]

**Comment by createyourpersonalaccount** [comment]
Is there something I still need to do? I'm not sure if @wargio did it for me or not.

**Comment by XVilka** [comment]
> Is there something I still need to do? I'm not sure if @wargio did it for me or not.

Yes, rebase and run formatter

**Comment by createyourpersonalaccount** [comment]
When I run

    git-clang-format-16 --extensions c,cpp,h,hpp,inc --style file dev

I get

    error: unknown key 'RequiresClausePosition'

on Debian bookworm. I had to remove that line to format the file.

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
One more formatting problem left:
```
Formatting [652/1630] librz/util/sdb/src/cdb_make.c
librz/util/sdb/src/cdb_make.c:13:18: error: code should be clang-formatted [-Wclang-format-violations]
#define ALIGNMENT sizeof(void *)
                 ^
```
Just use `sys/clang-format.py`

**Comment by createyourpersonalaccount** [comment]
Should https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#c be updated to mention the script instead?

**Comment by XVilka** [comment]
> Should https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#c be updated to mention the script instead?

Probably

**Comment by XVilka** [comment]
> When I run
> 
> ```
> git-clang-format-16 --extensions c,cpp,h,hpp,inc --style file dev
> ```
> 
> I get
> 
> ```
> error: unknown key 'RequiresClausePosition'
> ```
> 
> on Debian bookworm. I had to remove that line to format the file.

As for Debian, they seem to use old Clang/LLVM:



Debian 12bookworm/main | llvm-toolchain-15 | 15.0.6 | misc 
-- | -- | -- | --
Debian 12bookworm/main | llvm-toolchain-14 | 14.0.6 | misc 
Debian 12bookworm/main | llvm-toolchain-13 | 13.0.1 | misc 

(From https://repology.org/project/llvm/versions)

**Comment by createyourpersonalaccount** [comment]
@XVilka 
I installed `clang-format-16` from the llvm repo, which downloaded also its dependencies. Shouldn't it had worked?

**Comment by XVilka** [comment]
> Details

Ah, yes, then it should have worked. Probably some missing option, or the fact that formatting is often influenced by the surrounding code, which is not always available when it tries to format a patch instead, which `git-clang-format` does. At least from my understanding.

**Pull request merged**